packages/block-editor/src/store/selectors.js

```diff
@@ -2304,13 +2304,13 @@ function getUserPatterns( state ) {
 		return {
 			name: `core/block/${ userPattern.id }`,
 			id: userPattern.id,
-			title: userPattern.title.raw,
+			title: userPattern.title?.raw,
 			categories: userPattern.wp_pattern_category.map( ( catId ) =>
 				categories && categories.get( catId )
 					? categories.get( catId ).slug
 					: catId
 			),
-			content: userPattern.content.raw,
+			content: userPattern.content?.raw,
```

Comment:
I think `getUserPatterns` is not really used in the native version, so maybe we could skip updating it. WDYT?

Reply:
@fluiddot, I'm up for experimenting to see if the crash is resolved after skipping updating `getUserPatterns`, and only updating `buildReusableBlockInserterItem`. I've reverted that change in https://github.com/WordPress/gutenberg/pull/54792/commits/3e6923bd8f0d366f415f1cdd1054c2d5a8c46e1f.

The reason behind why I initially chose to update this function is that it's called in a couple of the selectors used on native. None of the sites I've looked at related to this crash actually have user-added reusable blocks, and I suspect there is a wider issue with patterns in general. If we continue to see this crash after this PR's merged, restoring that change could be a next step.